HaikuPorter/SourceFetcher.py

```diff
@@ -273,19 +290,27 @@ def __init__(self, uri, fetchTarget):
 		(unusedType, self.uri, self.rev) = parseCheckoutUri(uri)
 		if not self.rev:
 			self.rev = 'HEAD'
+		if self.rev.startswith('tag=') or self.rev.startswith('commit='):
+			self.rev=self.rev[self.rev.find('=') + 1:]
+			self.sourceShouldBeValidated = True
 
 	def fetch(self):
-		if not Configuration.shallAllowUnsafeSources():
-			sysExit('Downloading from unsafe sources is disabled in ' +
-					'haikuports.conf!')
+		if not self.sourceShouldBeValidated:
+			if not Configuration.shallAllowUnsafeSources():
+				sysExit('Downloading from unsafe sources is disabled in ' +
+						'haikuports.conf!')
 
-		warn("UNSAFE SOURCES ARE BAD AND SHOULD NOT BE USED IN PRODUCTION")
-		warn("PLEASE MOVE TO A STATIC ARCHIVE DOWNLOAD WITH CHECKSUM ASAP!")
+			warn("UNSAFE SOURCES ARE BAD AND SHOULD NOT BE USED IN PRODUCTION")
+			warn("PLEASE MOVE TO A STATIC ARCHIVE DOWNLOAD WITH CHECKSUM ASAP!")
```

Comment:
This isn't the case anymore I suppose, so we can remove all the "unsafe sources" logic? Or at least, we can remove it from this branch?

Reply:
Actually it would be more "PLEASE MOVE TO A TAG OR COMMIT WITH CHECKSUM!"

Reply:
Done